rdkit/Chem/RegistrationHash.py

```diff
@@ -176,38 +182,19 @@ def GetStereoTautomerHash(molecule):
   no_h_mol = _RemoveUnnecessaryHs(molecule)
   no_h_mol.UpdatePropertyCache(False)
 
+  cxflags = Chem.CXSmilesFields.CX_ATOM_LABELS | Chem.CXSmilesFields.CX_ENHANCEDSTEREO
```

Comment:
@irenazra can you provide a couple of examples where you saw things you didn't like that led to you needing to do this roundtrip?

Reply:
So, what I did not like is just the fact that I am doing this back and forth conversion mol to smiles back to mol, just to strip certain traits of the mol out. It feels like there is a better way for this. I think what we should do in the feature is to update the MolHash function itself to allow not only passing a boolean (useCxSmiles), but a way to pass in CXSmiles generation options/flags.

The reason why I had to this back and forth conversion to begin with is because I don't want the hash_with_cxExtensions to contain the extensions that we were stripping in GetCanonicalSmiles.

Reply:
https://github.com/rdkit/rdkit/issues/6224

Reply:
I agree that it would make more sense to resolve #6224 rather than to have this round-tripping code here.